src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T> {
```

Comment:
We should limit possible types in order not to give user create MachineService parametrized with Fruit, Animal etc
```suggestion
public interface MachineService<T extends Machine> {
```